tests/posix_semaphore/main.c

```diff
@@ -267,16 +274,11 @@ void test4(void)
         }
     }
 
-    fmt_u64_dec(uint64_str, elapsed);
+    uint64_str[fmt_u64_dec(uint64_str, elapsed)] = '\0';
     if (elapsed < (exp - 100)) {
```

Comment:
Maybe use the same margin here?

Reply:
Why ? if `native` is slower, there is no reason to allow a bigger 'before' timeout.